sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -6,72 +6,78 @@
+    SETTLEMENT_ABANDON,
+    SETTLEMENT_COMPLETE,
+    SETTLEMENT_DEFER,
+    SETTLEMENT_DEADLETTER,
+    ReceiveSettleMode,
+    _X_OPT_ENQUEUED_TIME,
+    _X_OPT_SEQUENCE_NUMBER,
+    _X_OPT_ENQUEUE_SEQUENCE_NUMBER,
+    _X_OPT_PARTITION_ID,
+    _X_OPT_PARTITION_KEY,
+    _X_OPT_VIA_PARTITION_KEY,
+    _X_OPT_LOCKED_UNTIL,
+    _X_OPT_LOCK_TOKEN,
+    _X_OPT_SCHEDULED_ENQUEUE_TIME
+)
+from ..exceptions import (
     MessageAlreadySettled,
-    MessageSettleFailed,
     MessageLockExpired,
-    SessionLockExpired)
+    SessionLockExpired,
+    MessageSettleFailed
+)
+from .utils import utc_from_timestamp, utc_now
 
 
 class Message(object):  # pylint: disable=too-many-public-methods,too-many-instance-attributes
     """A Service Bus Message.
 
-    :param body: The data to send in a single message. The maximum size per message is 256 kB.
+    :param body: The data to send in a single message.
     :type body: str or bytes
-    :param encoding: The encoding for string data. Default is UTF-8.
-    :type encoding: str
+    :param str encoding: The encoding for string data. Default is UTF-8.
+    :keyword session_id: An optional session ID for the message to be sent.
+    :paramtype session_id: str
 
     .. admonition:: Example:
-        .. literalinclude:: ../samples/sync_samples/test_examples.py
+
+        .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
             :start-after: [START send_complex_message]
             :end-before: [END send_complex_message]
             :language: python
             :dedent: 4
             :caption: Sending a message with additional properties
 
-        .. literalinclude:: ../samples/sync_samples/test_examples.py
-            :start-after: [START receive_complex_message]
-            :end-before: [END receive_complex_message]
-            :language: python
-            :dedent: 4
-            :caption: Checking the properties on a received message
-
     """
 
-    _X_OPT_ENQUEUED_TIME = b'x-opt-enqueued-time'
-    _X_OPT_SEQUENCE_NUMBER = b'x-opt-sequence-number'
-    _X_OPT_ENQUEUE_SEQUENCE_NUMBER = b'x-opt-enqueue-sequence-number'
-    _X_OPT_PARTITION_ID = b'x-opt-partition-id'
-    _X_OPT_PARTITION_KEY = b'x-opt-partition-key'
-    _X_OPT_VIA_PARTITION_KEY = b'x-opt-via-partition-key'
-    _X_OPT_LOCKED_UNTIL = b'x-opt-locked-until'
-    _x_OPT_LOCK_TOKEN = b'x-opt-lock-token'
-    _x_OPT_SCHEDULED_ENQUEUE_TIME = b'x-opt-scheduled-enqueue-time'
-
     def __init__(self, body, encoding='UTF-8', **kwargs):
         subject = kwargs.pop('subject', None)
-        # Although we might normally thread through **kwargs this causes problems as MessageProperties won't absorb spurious args.
+        # Although we might normally thread through **kwargs this causes
+        # problems as MessageProperties won't absorb spurious args.
         self.properties = uamqp.message.MessageProperties(encoding=encoding, subject=subject)
         self.header = uamqp.message.MessageHeader()
```

Comment:
Can you make sure all these public attributes are documented in the class docstring as instance variables (`:ivar:`)

Reply:
added!